dms/models/dms_file.py

```diff
@@ -0,0 +1,631 @@
+            res_vals["res_id"] = attachment.res_id
+            del res_vals["content"]
+        return res_vals
+
+    def copy(self, default=None):
+        self.ensure_one()
+        default = dict(default or [])
+        if "directory_id" in default:
+            model = self.env["dms.directory"]
+            directory = model.browse(default["directory_id"])
+            names = directory.sudo().file_ids.mapped("name")
+        else:
+            names = self.sudo().directory_id.file_ids.mapped("name")
+        default.update({"name": file.unique_name(self.name, names, self.extension)})
+        return super(File, self).copy(default)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_vals_list = []
+        for vals in vals_list:
+            if "attachment_id" not in vals:
+                vals = self._create_model_attachment(vals)
+            new_vals_list.append(vals)
+        return super(File, self).create(new_vals_list)
+
+    # ----------------------------------------------------------
+    # Locking fields and functions
+    # ----------------------------------------------------------
+
+    locked_by = fields.Many2one(comodel_name="res.users")
+
+    is_locked = fields.Boolean(compute="_compute_locked", string="Locked")
+
+    is_lock_editor = fields.Boolean(compute="_compute_locked", string="Editor")
+
+    # ----------------------------------------------------------
+    # Locking
+    # ----------------------------------------------------------
+
+    def lock(self):
+        self.write({"locked_by": self.env.uid})
+
+    def unlock(self):
+        self.write({"locked_by": None})
+
+    # ----------------------------------------------------------
+    # Read, View
+    # ----------------------------------------------------------
+
+    @api.depends("locked_by")
+    def _compute_locked(self):
+        for record in self:
+            if record.locked_by.exists():
+                record.update(
+                    {
+                        "is_locked": True,
+                        "is_lock_editor": record.locked_by.id == record.env.uid,
+                    }
+                )
+            else:
+                record.update({"is_locked": False, "is_lock_editor": False})
+
+    def get_attachment_object(self, attachment):
+        return {
+            "name": attachment.name,
+            "datas": attachment.datas,
+            "res_model": attachment.res_model,
+            "mimetype": attachment.mimetype,
+        }
+
+    def get_dms_files_from_attachments(self, attachment_ids=None, view_type="list"):
+        """Get the dms files from uploaded attachments.
+        :return: An Array of dms files.
+        """
+
+        if attachment_ids is None:
+            attachment_ids = []
+        attachments = self.env["ir.attachment"].browse(attachment_ids)
+
+        if not attachments:
+            raise UserError(_("No attachment was provided"))
```

Comment:
```suggestion
    def get_dms_files_from_attachments(self, attachment_ids=None):
        """Get the dms files from uploaded attachments.
        :return: An Array of dms files.
        """
        if not attachments:
            raise UserError(_("No attachment was provided"))

        attachments = self.env["ir.attachment"].browse(attachment_ids)
```

A little nitpick that simplifies things a bit; if no attachments are given it will raise anyway. Also the kwarg is somewhat misleading so better to drop it.

Reply:
great and thanks! Your suggestion has been adjusted